addons/mail/static/src/core/common/store_service.js

```diff
@@ -203,6 +203,22 @@ export class Store extends BaseStore {
         },
     });
 
+    standaloneInboxMessages = Record.many("mail.message", {
+        compute() {
+            const tab = this.store.discuss.activeTab;
+            if (tab !== "main") {
+                return [];
+            }
```

Comment:
Global state on a field such as this one shouldn't depend on the state of the UI.
It's also counter-productive to force a recompute every time the tab is opened.